File: src/main/java/com/grosner/dbflow/sql/migration/Migration.java
package com.grosner.dbflow.sql.migration;

import android.database.sqlite.SQLiteDatabase;

/**
 * Author: andrewgrosner
 * Contributors: { }
 * Description: Called when the Database is migrating. We can perform custom migrations here.
 */
public interface Migration {

    /**
     * Returns the DB migration version
     * @return
     */
    public int getNewVersion();

    /**
     * Called before we migrate data
     */
    public void onPreMigrate();

    /**
     * Perform your migrations here
     * @param database
     */
    public void migrate(SQLiteDatabase database);

    /**
     * Called after the migration completes
     */
    public void onPostMigrate();
}

File: dbflow/src/main/java/com/raizlabs/android/dbflow/migration/Migration.kt
package com.raizlabs.android.dbflow.migration

import com.raizlabs.android.dbflow.database.DatabaseWrapper

/**
 * Description: Called when the Database is migrating. We can perform custom migrations here. A [com.raizlabs.android.dbflow.annotation.Migration]
 * is required for registering this class to automatically be called in an upgrade of the DB.
 */
interface Migration {

    /**
     * Called before we migrate data. Instantiate migration data before releasing it in [.onPostMigrate]
     */
    fun onPreMigrate()

    /**
     * Perform your migrations here
     *
     * @param database The database to operate on
     */
    fun migrate(database: DatabaseWrapper)

    /**
     * Called after the migration completes. Release migration data here.
     */
    fun onPostMigrate()
}
